rewriter/Minitest.cc

```diff
@@ -653,6 +654,99 @@ ast::ExpressionPtr runSingle(core::MutableContext ctx, bool isClass, ast::Send *
+            private:
+                core::NameRef attributeName;
+                core::NameRef isExpectedName;
+                core::NameRef expectName;
+
+            public:
+                IsExpectedTransformer(core::NameRef attributeName, core::NameRef isExpectedName,
+                                      core::NameRef expectName)
+                    : attributeName(attributeName), isExpectedName(isExpectedName), expectName(expectName) {}
+
+                ast::ExpressionPtr postTransformSend(core::Context ctx, ast::ExpressionPtr tree) {
+                    auto send = ast::cast_tree<ast::Send>(tree);
+                    if (send == nullptr) {
+                        return tree;
+                    }
+
+                    // Look for is_expected calls
+                    if (send->fun == isExpectedName && send->recv.isSelfReference()) {
+                        // Replace is_expected with expect(subject.attribute)
+                        auto subjectCall = ast::MK::Send0(send->loc, ast::MK::Self(send->loc), core::Names::subject(),
+                                                          send->loc.copyWithZeroLength());
+                        auto attributeCall = ast::MK::Send0(send->loc, std::move(subjectCall), attributeName,
+                                                            send->loc.copyWithZeroLength());
+                        return ast::MK::Send1(send->loc, ast::MK::Self(send->loc), expectName,
+                                              send->loc.copyWithZeroLength(), std::move(attributeCall));
+                    }
+
+                    return tree;
+                }
+            };
+
+            ast::ExpressionPtr itBody = std::move(block->body);
+            IsExpectedTransformer transformer(attributeName, isExpectedName, expectName);
+            itBody = ast::TreeMap::apply(ctx, transformer, std::move(itBody));
+
+            // Create it block
+            auto itName = ctx.state.enterNameUTF8("<it>");
+            auto itDeclLoc = send->loc.copyWithZeroLength();
+
+            ConstantMover constantMover;
+            ast::TreeWalk::apply(ctx, constantMover, itBody);
+
+            auto itMethod = ast::MK::SyntheticMethod0(send->loc, itDeclLoc, itName,
+                                                      prepareBody(ctx, /* isClass */ true, std::move(itBody),
+                                                                  /* insideDescribe */ true));
+            ast::cast_tree_nonnull<ast::MethodDef>(itMethod).flags.discardDef = true;
+            itMethod = addSigVoid(ctx, move(itMethod));
+            itMethod = constantMover.addConstantsToExpression(send->loc, move(itMethod));
+
+            // No need to create a subject method - the body now calls subject.attribute directly
+            ast::ClassDef::RHS_store describeBody;
+            describeBody.emplace_back(std::move(itMethod));
+
+            // Create describe block containing the subject and it methods
```

Comment:
- we're creating a class, not a describe block
- the thing we're creating does not contain a subject

Reply:
721f267f1